focus-android/quality/detekt.yml

```diff
@@ -596,15 +687,15 @@ style:
   UnusedPrivateMember:
     active: true
     allowedNames: '(_|ignored|expected|serialVersionUID)'
-    ignoreAnnotated: ['Composable']
+    ignoreAnnotated: ['Preview'] # Configuration for Compose https://github.com/mozilla-mobile/android-components/issues/11866
   UseAnyOrNoneInsteadOfFind:
-    active: false
+    active: true
   UseArrayLiteralsInAnnotations:
```

Comment:
Same for this one, [the docs](https://detekt.dev/docs/rules/style/#usearrayliteralsinannotations) say it's more readable but I don't see any reason to make us change our default.

Reply:
This is enabled by default in detekt since v1.21.0 https://detekt.dev/docs/rules/style/#useanyornoneinsteadoffind. I wouldn't say we really have our own default except where we have a comment that stated that we enabled or changed a property and the PR/issue that was responsible for that decision. Our default is more based on what was the default of the previous base configs. 

I would lean towards following the base configuration because keeping this inactive means we have an intentional intervention and now requires another comment about why or when we made this decision. Looking at the code sample, I don't see any reason why we wouldn't want to accept more readable code. 

I am also thinking about this from an upgrade point of view where we are currently diff our current config and the new base config to sync any new rules that needs to be added to the file.

Reply:
I am also looking to turn on more rules after this. Especially rules that will make our code more readable and simplifies things for us. Based on our review guidelines, I want to front load and really empower our linters to do most of the heavy lifting because most of my review codes are based around simplifying and making code more readable.

Reply:
> This is enabled by default in detekt since v1.21.0 https://detekt.dev/docs/rules/style/#useanyornoneinsteadoffind. 

I was referring to the line I commented on, [`UseArrayLiteralsInAnnotations`](https://detekt.dev/docs/rules/style/#usearrayliteralsinannotations). Using `any` or `none` sounds fine to me.

> Our default is more based on what was the default of the previous base configs.

This is not true. We chose our defaults intentionally based on our needs. 

> Looking at the code sample, I don't see any reason why we wouldn't want to accept more readable code.

I am all for this, but using defaults aren't adding that value, it is just code churn. `UndocumentedPublicClass`, `UnreachableCatchBlock` and `CognitiveComplexMethod` are _valuable_ to the code base. The detekt defaults are made by the developers that want to switch some changes to true because they invested in building those but that does not mean those make sense for us.

> making code more readable.

Great! Some of these checks are not readable to me though (and maybe others). 😅